### PR TITLE
Path as module

### PR DESCRIPTION
**Pull request by Kheprys**
Define Path as a module and cleanup unused functions

tested in simulation for both _cup2019_ and _app_test_ targets.

### CODE REVIEW

File: lib/path/include/path.h [hunk 1]
```diff
@@ -1,15 +1,46 @@
+/*
+ * Copyright (C) 2021 COGIP Robotics association <cogip35@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    lib_path Containing Path structure definition and utilities function to work with path
```
Comment (ecourtois) [inline lib/path/include/path.h:10]: utility functions

File: lib/path/include/path.h [hunk 2]
```diff
@@ -1,15 +1,46 @@
...
 typedef struct {
     pose_t pos;
     uint8_t allow_reverse;
     double max_speed;
     func_cb_t act;
 } path_pose_t;
 
+/**
+ * @brief Path type
+ */
```
Comment (ecourtois) [inline lib/path/include/path.h:43]: nb_pose**s**?

File: lib/path/include/path.h [hunk 3]
```diff
@@ -20,15 +51,73 @@ typedef struct {
...
+ * @brief Global variable containing complete robot path
+ */
 extern path_t robot_path;
 
+/**
+ * @brief Get current position
+ * @param[in]    path    robot path
+ * @return               current position of the path
+ */
 const path_pose_t *path_get_current_path_pos(const path_t *path);
```
Comment (ecourtois) [inline lib/path/include/path.h:64]: path_get_current_pose(...)?

File: lib/path/include/path.h [hunk 4]
```diff
@@ -20,15 +51,73 @@ typedef struct {
...
+
+/**
+ * @brief Get position at given index
+ * @param[in]    path    robot path
+ * @param[in]    idx     index to get the position from
+ * @return               position at the given index
+ */
 const path_pose_t *path_get_pose_at_idx(path_t *path, uint8_t idx);
```
Comment (ecourtois) [inline lib/path/include/path.h]: const path_pose_t *path_get_pose(**const** path_t *path, uint8_t idx);

File: lib/path/include/path.h [hunk 5]
```diff
@@ -20,15 +51,73 @@ typedef struct {
...
+
+/**
+ * @brief Get index of the current position
+ * @param[in]    path    robot path
+ * @return               index of the current position
+ */
 uint8_t path_get_current_pose_idx(path_t *path);
```
Comment (ecourtois) [inline lib/path/include/path.h]: const path_t *

File: lib/path/include/path.h [hunk 6]
```diff
@@ -20,15 +51,73 @@ typedef struct {
...
+
+/**
+ * @brief Get maximum speed to use in order to go to the current position
+ * @param[out]    path    robot path
+ * @return               maximum speed to use to go to the current position
+ */
 uint8_t path_get_current_max_speed(const path_t *path);
+
+/**
+ * @brief Mirror all points in path to match the two possible side of the game
```
Comment (ecourtois) [inline lib/path/include/path.h:92]: side**s**

**Review by ecourtois — CHANGES_REQUESTED** [review]
Almost good, I would squash both commits to make it easier to review.

**Review by ecourtois — APPROVED** [review]
LGTM

File: applications/app_test/path.c [hunk 7]
```diff
@@ -47,74 +47,6 @@ path_t robot_path = {
     /* static cfg */
     .current_pose_idx = 0,
     .play_in_loop = TRUE,
-    .nb_pose = sizeof(poses) / sizeof(path_pose_t),
+    .nb_poses = sizeof(poses) / sizeof(path_pose_t),
```
Comment (gdoffe) [inline applications/app_test/path.c:50]: Please use ARRAY_SIZE instead.

**Review by gdoffe — APPROVED** [review]
LGTM

**Pull request merged**